### PR TITLE
Add optional services and applications to release markers

### PR DESCRIPTION
**Pull request by timoinstana**
# Why

The upcoming release of Instana will feature the possibility to have scoped release markers. The supported scopes are `applications` and `service`.

# What

Add the two new optional parameters named `services` and `applications` to ReleaseMarker and ReleaseMarkerStep. Add test cases which cover cases with services and/or applications set.

Example usage with scopes:
```
releaseMarker releaseName: "Release 4711", services: [service(name: "my-service-1"), service(name: "my-service-2")]
releaseMarker releaseName: "Release 4711", applications: [application(name: "My Application")]
```

### CODE REVIEW

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java [hunk 1]
```diff
@@ -48,8 +49,26 @@ static HttpRequestExecution from(ReleaseMarker http, TaskListener taskListener)
...
+				JSONObject service = new JSONObject();
+				service.put("name", serviceName);
+				services.add(service);
+			}
+			jsonObject.put("services", services);
+		}
+		if (http.getApplicationNames() != null) {
+			List<JSONObject> applications = new ArrayList<>();
+			for (String applicationName : http.getApplicationNames()) {
+				JSONObject service = new JSONObject();
```
Comment (cedricziel) [inline src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Nitpick: `application`

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java [hunk 2]
```diff
@@ -48,8 +49,26 @@ static HttpRequestExecution from(ReleaseMarker http, TaskListener taskListener)
...
+				service.put("name", applicationName);
+				applications.add(service);
+			}
+			jsonObject.put("applications", applications);
+		}
 		String body = jsonObject.toString();
-
+		System.out.println(body);
```
Comment (cedricziel) [inline src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: I think we should drop this one

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java [hunk 3]
```diff
@@ -62,7 +81,26 @@ static HttpRequestExecution from(ReleaseMarkerStep step, TaskListener taskListen
...
+				JSONObject service = new JSONObject();
+				service.put("name", serviceName);
+				services.add(service);
+			}
+			jsonObject.put("services", services);
+		}
+		if (step.getApplicationNames() != null) {
+			List<JSONObject> applications = new ArrayList<>();
+			for (String applicationName : step.getApplicationNames()) {
+				JSONObject service = new JSONObject();
```
Comment (cedricziel) [inline src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Nitpick: `application`

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java [hunk 4]
```diff
@@ -62,7 +81,26 @@ static HttpRequestExecution from(ReleaseMarkerStep step, TaskListener taskListen
...
+				service.put("name", applicationName);
+				applications.add(service);
+			}
+			jsonObject.put("applications", applications);
+		}
 		String body = jsonObject.toString();
+		System.out.println(body);
```
Comment (cedricziel) [inline src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Can we drop this as well?

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java (hunk 2 above)
Comment (cedricziel) [reply src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Or is this used for the assertions? If yes, we should either look for an alternative or at least comment so we know why it's here :)

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java (hunk 1 above)
Comment (timoinstana) [reply src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Good catch from copy/pasting. Changed to `application`.

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java (hunk 2 above)
Comment (timoinstana) [reply src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Yes, we should remove it from here.

File: src/main/java/jenkins/plugins/instana/HttpRequestExecution.java (hunk 4 above)
Comment (timoinstana) [reply src/main/java/jenkins/plugins/instana/HttpRequestExecution.java]: Yep

**Review by cedricziel — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -30,3 +30,22 @@ If you wish to create a release for a particular timestamp you can use the optio
 ```groovy
 releaseMarker releaseName: "Test Release ${currentBuild.number}", releaseStartTimestamp: "1564486446000"
 ```
+
+To create a release scoped to a given service or application you can add the `services` or `applications` as parameter
```
Comment (YoannBuch) [inline README.md:34]: @timoinstana is service scoped to AP supported?

**Pull request merged**